CMakeLists.txt

```diff
@@ -13,6 +13,13 @@ ENDIF()
 # This project name is cyclus.
 PROJECT(CYCLUS VERSION 1.6.0)
 
+SET(CYCLUS_CXX_LARGE_INT "std::numeric_limits<int>::max()")
+SET(CYCLUS_CXX_LARGE_DOUBLE "1e299")
+SET(CYCLUS_CXX_NEAR_ZERO "1e-08")
+SET(CYCLUS_PYTHON_LARGE_INT "(2 ** (ctypes.sizeof(ctypes.c_int) * 8 - 1)) - 1")
+SET(CYCLUS_PYTHON_LARGE_DOUBLE "1e299")
+SET(CYCLUS_PYTHON_NEAR_ZERO "1e-08")
```

Comment:
Thanks for pursuing this approach... do we need to define both CXX and PYTHON versions or can we just have one version that we use in both places?

Reply:
Oh - I see now that these are strings... so the LARGE_INT needs to be different, but not the others, right?

Reply:
Correct, just the INTs have different values. I defined two complete sets for the sake of consistency and transparency, but we can just have two versions of CYCLUS_LARGE_INT if that feels cleaner or more maintainable. I don't have a strong preference either way

Reply:
It seems useful for there to be a way to ensure that the numerical value is the same in all cases.

For the non-INTs, we could keep two variables, but define one with the other.

For the INTS - I'm not sure what our options are, short of defining it with some constant rather than an executable code snippet.

Reply:
We can use CMake to check the size of int and set the constant appropriately - this could replace the quoted snippet above:

```cmake
INCLUDE(CheckTypeSize)
CHECK_TYPE_SIZE("int" INT_SIZE_IN_BYTES LANGUAGE CXX)
MATH(EXPR INT_MAX "(1 << (${INT_SIZE_IN_BYTES} * 8 - 1)) - 1")

SET(CYCLUS_LARGE_INT "${INT_MAX}")
SET(CYCLUS_LARGE_DOUBLE "1e299")
SET(CYCLUS_NEAR_ZERO "1e-08")
```

Reply:
This looks like a good solution - thanks @bennibbelink